tools/file_packager.py

```diff
@@ -191,17 +198,13 @@
 if not has_preloaded or jsoutput == None:
   assert not separate_metadata, 'cannot separate-metadata without both --preloaded files and a specified --js-output'
 
+# `"__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__"` will be replaced with actual code on later stage of the build, this way Closure Compiler will not mangle it
```

Comment:
Overall this PR looks good - is it waiting on anything else?

One last thing, please add the `__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__` string as a constant in `tools/shared.py` (perhaps next to `global_initializers_pattern`). `shared.py` is imported by the two files that need this (emcc and file packager), so that way we can avoid duplicating the string contents (risk of a typo).